avocado/utils/pci.py

```diff
@@ -0,0 +1,310 @@
+import os
+from . import process
+
+
+def get_domains():
+    """
+    Gets all pci domains.
+
+    :return: List of pci domains.
+    """
+    cmd = "lspci -D"
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if output:
+        domains = []
+        for line in output.splitlines():
+            domains.append(line.split(":")[0])
+        return list(set(domains))
+
+
+def get_num_devices_in_domain(domain):
+    """
+    Gets number of devices in a pci domain.
+
+    :parm domain: pci domain.
+
+    :return: number of devices in a pci domain.
+    """
+    cmd = "ls -l /sys/class/*/ -1"
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if output:
+        count = 0
+        for line in output.splitlines():
+            if domain in line:
+                count += 1
+        return count
+
+
+def get_disks_in_device(device):
+    """
+    Gets disks in a pci device.
+
+    :parm device: pci device.
+
+    :return: list of disks in a pci device.
+    """
+    disks_path = "/dev/disk/by-path/"
+    devices_list = []
+    for dev in os.listdir(disks_path):
+        if device in dev:
+            link = os.readlink("%s%s" % (disks_path, dev))
+            devices_list.append(os.path.abspath("%s%s" % (disks_path, link)))
+    return devices_list
+
+
+def get_interfaces_in_device(device):
+    """
+    Gets ethernet interface in a pci device.
+
+    :parm device: pci device.
+
+    :return: list of interfaces in a pci device.
+    """
+    iface_path = "/sys/class/net/"
+    interfaces_list = []
+    for iface in os.listdir(iface_path):
+        if device in os.readlink("%s%s" % (iface_path, iface)):
+            interfaces_list.append(iface)
+    return interfaces_list
+
+
+def get_pci_fun_list(device):
+    """
+    Gets list of functions in the given pci device.
+    Example: in device 0000:03:00, functions are 0000:03:00.0 and 0000:03:00.1
+
+    :parm device: pci device.
+
+    :return: list of functions in a pci device.
+    """
+    return list(dev for dev in get_pci_devices() if device in dev)
+
+
+def get_slot_from_sysfs(device):
+    """
+    Gets the pci slot of given device.
+    Specific for ppc64 processor
+
+    :parm device: pci device.
+
+    :return: slot of pci device from sysfs.
+    """
+    if not os.path.isdir('/sys/bus/pci/devices/%s' % device):
+        return
+    cmd = "cat /sys/bus/pci/devices/%s/devspec" % device
+    devspec = process.system_output(cmd, shell=True).strip('\n')
+    cmd = "cat /proc/device-tree/%s/ibm,loc-code" % devspec
+    slot = process.system_output(cmd, shell=True).strip('\0\n')
+    return re.match(r'((\w+)[\.])+(\w+)-(\w*\d+)-(\w*\d+)|Slot(\d+)',
+                    slot).group()
+
+
+def get_pci_id_from_sysfs(device):
+    """
+    Gets the pci id from sysfs of given device.
+
+    :parm device: pci device.
+
+    :return: pci id of a pci device from sysfs.
+    """
+    path = "/sys/bus/pci/devices/%s" % device
+    if os.path.isdir(path):
```

Comment:
Not really an issue, but wouldn't this looks nicer?

```python
path = "/sys/bus/pci/devices/%s/%%s" % device
try:
    return ":".join("%04x" % int(open(path % param).read(), 16)
                    for param in ["vendor", "device", "subsystem_vendor"
                                  "subsystem_device"]])
except IOError:    # return None when some file does not exists
    return None
```

PS: It's just a matter of taste, some prefer this, others that, no need to change it, I put it here only for your consideration and if you like it, you can adopt this style. If not, then keep it as is.